### PR TITLE
[runtime env] Clarify error message about where to install `smart_open` for remote URI

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
At least two users reported encountering 
```
ImportError(
                            "You must `pip install smart_open` and "
                            "`pip install boto3` to fetch URIs in s3 "
                            "bucket. "
```
and trying to fix it by specifying them in the `pip` field of `runtime_env`, which won't work because the runtime_env setup code doesn't run inside the runtime_env.  This PR clarifies the error message to say that they must be preinstalled on the cluster, and adds a note to the docs.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-core/handling-dependencies.rst [hunk 1]
```diff
@@ -572,7 +572,8 @@ Currently, three types of remote URIs are supported for hosting ``working_dir``
 
     - ``runtime_env = {"working_dir": "gs://example_bucket/example_file.zip"}``
 
-
+Note that the packages ``smart_open``, ``boto3`` and ``google-cloud-storage`` are not installed by default, and it is not sufficient to specify them in the ``pip`` section of your ``runtime_env``.
```
Comment (angelinalg) [inline doc/source/ray-core/handling-dependencies.rst:575]:
```suggestion
Note that the ``smart_open``, ``boto3``, and ``google-cloud-storage`` packages are not installed by default, and it is not sufficient to specify them in the ``pip`` section of your ``runtime_env``.
```

**Review by rkooo567 — APPROVED** [review]

File: python/ray/_private/runtime_env/packaging.py [hunk 2]
```diff
@@ -683,7 +689,7 @@ def open_file(uri, mode, *, transport_params=None):
                     except ImportError:
                         raise ImportError(
                             "You must `pip install smart_open` "
-                            f"to fetch {protocol.value.upper()} URIs."
+                            f"to fetch {protocol.value.upper()} URIs." + install_warning
```
Comment (shrekris-anyscale) [inline python/ray/_private/runtime_env/packaging.py:692]:
```suggestion
                            f"to fetch {protocol.value.upper()} URIs. " + install_warning
```

**Review by shrekris-anyscale — APPROVED** [review]

**Review by angelinalg — APPROVED** [review]
LGTM

**Comment by architkulkarni** [comment]
Test failures:
- serve:tutorial_rllib broken on master, unrelated
- tests:test_task_metrics flaky on master, unrelated
- Doc build failure "to_summary_by_lineage" unrelated and was broken on master
- air:test_remote_storage_hdfs broken on master, unrelated
- multi_agent_independent_learning broken on master, unrelated
- arm wheel build "Observed wheel commit () is not expected commit" happens on master sometimes, flaky and unrelated

**Pull request merged**